code/tests/NFluent.Tests/DateTimeOffsetRelatedTests.cs

```diff
@@ -411,5 +412,55 @@ public void NotMatchTheSameUtcInstantWorks()
 
             Check.That(oneHourOffSet).Not.MatchTheSameUtcInstantIgnoringMillis(new DateTimeOffset(localTime, new TimeSpan(6, 0, 0)));
         }
+        
+        [Test]
+        public void IsCloseToTimeSpanWorks()
+        {
+            var reference = DateTimeOffset.UtcNow;
+            
+            Check.That(reference).IsCloseTo(reference.AddMilliseconds(1), TimeSpan.FromMilliseconds(1));
+            Check.That(reference).IsCloseTo(reference.AddMilliseconds(-1), TimeSpan.FromMilliseconds(1));
+        }
+        
+        [Test]
+        public void IsCloseToDurationWorks()
+        {
+            var reference = DateTimeOffset.UtcNow;
+            
+            Check.That(reference).IsCloseTo(reference.AddMilliseconds(1), new Duration(1, TimeUnit.Milliseconds));
+            Check.That(reference).IsCloseTo(reference.AddMilliseconds(-1), new Duration(1, TimeUnit.Milliseconds));
+        }
+
+        [Test]
+        public void IsCloseToTimeSpanShouldFailsIfToFar()
+        {
+            var reference = DateTimeOffset.UtcNow;
+            var expected = reference.AddMilliseconds(2);
+            var within = TimeSpan.FromMilliseconds(1);
+
+            Check.ThatCode(() => Check.That(reference).IsCloseTo(expected, within))
+                 .IsAFailingCheckWithMessage("",
+                                             "The checked date time is outside the expected value range.",
+                                             "The checked date time:",
+                                             "\t[" + reference.ToStringProperlyFormatted() + "]",
+                                             "The expected value:",
+                                             "\t[" + expected + " (+/- " + within + ")]");
+        }
+        
+        [Test]
+        public void IsCloseToDurationShouldFailsIfToFar()
+        {
+            var reference = DateTimeOffset.UtcNow;
+            var expected = reference.AddMilliseconds(2);
+            var within = new Duration(1, TimeUnit.Milliseconds);
+
+            Check.ThatCode(() => Check.That(reference).IsCloseTo(expected, within))
+                 .IsAFailingCheckWithMessage("",
```

Comment:
You should use a constant date so that the error message is a constant! In this case, it would highlight that TimeSpan does not provide a simple (to read) output.